test/FastTests/Sparrow/Hashing.cs

```diff
@@ -286,215 +234,286 @@ public unsafe void EnsureZeroLengthStringIsAValidHash()
                 Assert.Equal(marvinHash, Hashing.Marvin32.CalculateInline(new List<int>()));
             }
         }
 
-
-
-
-        public static IEnumerable<object[]> BufferSize
-        {
-            get
-            {
-                return new[]
-                {
-                    new object[] {1},
-                    new object[] {4},
-                    new object[] {15},
-                    new object[] {65},
-                    new object[] {90},
-                    new object[] {128},
-                    new object[] {129},
-                    new object[] {1000},
-                };
-            }
-        }
-
-        [Theory]
-        [MemberData("BufferSize")]
-        public void XXHash32_IterativeHashingEquivalence(int bufferSize)
-        {
-            var rnd = new Random(1000);
-
-            byte[] values = new byte[bufferSize];
-            rnd.NextBytes(values);
-
-            uint seed = 233;
-            var context = Hashing.Iterative.XXHash32.Preprocess(values, values.Length, seed);
-
-            for (int i = 1; i < values.Length; i++)
-            {
-                var expected = Hashing.XXHash32.Calculate(values, i, seed);
-                var result = Hashing.Iterative.XXHash32.Calculate(values, i, context);
-                Assert.Equal(expected, result);
-            }
-        }
-
-        [Theory]
-        [MemberData("BufferSize")]
-        public void XXHash32_IterativeHashingPrefixing(int bufferSize)
-        {
-            var rnd = new Random(1000);
-
-            byte[] values = new byte[bufferSize];
-            rnd.NextBytes(values);
-
-            uint seed = 233;
-            var context = Hashing.Iterative.XXHash32.Preprocess(values, values.Length, seed);
-
-            for (int i = 1; i < values.Length; i++)
-            {
-                var expected = Hashing.XXHash32.Calculate(values, i, seed);
-
-                for (int j = 0; j < i; j++)
-                {
-                    var result = Hashing.Iterative.XXHash32.Calculate(values, i, context, j);
-                    Assert.Equal(expected, result);
-                }
-            }
-        }
-
-        [Fact]
+        [RavenFact(RavenTestCategory.Core)]
         public void Combine()
         {
             int h1 = Hashing.HashCombiner.CombineInline(1991, 13);
             int h2 = Hashing.HashCombiner.CombineInline(1991, 12);
             Assert.NotEqual(h1, h2);
         }
 
-
-        [Theory]
-        [MemberData("BufferSize")]
-        public unsafe void XXHash32_StreamedHashingEquivalence(int bufferSize)
+        [RavenTheory(RavenTestCategory.Core)]
+        [InlineDataWithRandomSeed]
+        public unsafe void XXHash64_StreamedHashingEquivalence(int randomSeed)
         {
-            var rnd = new Random(1000);
+            var rnd = new Random(randomSeed);
+            var bufferSize = rnd.Next(1, 10000);
+            var seed = (ulong)rnd.Next(1, 10000);
```

Comment:
Do we need to limit it to be in a range (1, 10000)?

Reply:
I went ahead and shrink it a bit and allowed the seed to be whatever `ulong/uint`